src/ads/ads-controller.js

```diff
@@ -0,0 +1,68 @@
+//@flow
+import Player from '../player';
+import {AdEventType} from './ad-event-type';
+import EventManager from '../event/event-manager';
+import FakeEventTarget from '../event/fake-event-target';
+import FakeEvent from '../event/fake-event';
+import {CustomEventType} from '../event/event-type';
+import Error from '../error/error';
+
+class AdsController extends FakeEventTarget implements IAdsController {
+  _player: Player;
+  _adsPluginController: IAdsController;
+  _allAdsCompleted: boolean;
+  _eventManager: EventManager;
+
+  /**
+   * The ads controller.
+   * @param {Player} player - The player.
+   * @param {IAdsController} adsPluginController - the controller of the current plugin instance.
+   */
+  constructor(player: Player, adsPluginController: IAdsController) {
```

Comment:
doesn't seem to use both `player` and `adsPluginController`

Reply:
`player` used for the events. `adsPluginController` used for ads API e.g `adsPluginController.skipAd()`